### PR TITLE
Mepts 906 | List of TPT Eligibility Changes September Release

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenOnARTandFormulationsDataset.java [hunk 1]
```diff
@@ -218,7 +218,7 @@ private DataDefinition getAge() {
     return spdd;
   }
 
-  private DataDefinition getNID() {
+  public DataDefinition getNID() {
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenOnARTandFormulationsDataset.java]: Currently this column is NOT specifying the identifier type, that need to be provided for the correct  identifier value. As things stand now, any identifier will be pulled. See how I did that https://github.com/jembi/openmrs-module-eptsreports/blob/MEPTS-903-Data_Quality_Report_to_Identify_Duplicate/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/DQRForDuplicateFichaResumoDataSet.java#L123-L139, you might want to use that or make changes to your code to include those

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TPTListOfPatientsEligibleDataSet.java [hunk 2]
```diff
@@ -44,6 +44,9 @@
   @Autowired private TPTEligiblePatientListCohortQueries tPTEligiblePatientListCohortQueries;
   @Autowired private HivMetadata hivMetadata;
 
+  @Autowired
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TPTListOfPatientsEligibleDataSet.java:45]: Can you please initialize those variables in a public constructor as opposed to having separate @Autowired?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TPTListOfPatientsEligibleDataSet.java [hunk 3]
```diff
@@ -66,7 +69,7 @@ public DataSetDefinition constructDataset() throws EvaluationException {
 
     pdd.addColumn("id", new PersonIdDataDefinition(), "");
     pdd.addColumn("name", nameDef, "");
-    pdd.addColumn("nid", identifierDef, "");
+    pdd.addColumn("nid", listChildrenOnARTandFormulationsDataset.getNID(), "");
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TPTListOfPatientsEligibleDataSet.java]: Make sure an identifier type is factored in

**Review by ningosi — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenOnARTandFormulationsDataset.java (hunk 1 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenOnARTandFormulationsDataset.java]: Once ticket 903 is not merged, I made changes on the code to include this identifier value.

**Review by ningosi — APPROVED** [review]

**Pull request merged**